plugins/woocommerce/client/admin/client/marketplace/components/product-preview-modal/product-preview-modal.tsx

```diff
@@ -0,0 +1,191 @@
+/**
+ * External dependencies
+ */
+import { Modal } from '@wordpress/components';
+import { useEffect, useState, useCallback } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { Spinner } from '@woocommerce/components';
+
+/**
+ * Internal dependencies
+ */
+import { fetchProductPreview } from '../../utils/functions';
+import sanitizeHtmlExtended from '~/lib/sanitize-html/sanitize-html-extended.js';
+import sanitizeHtmlConfig from './product-preview-sanitize-html-config';
+import './product-preview-modal.scss';
+
+interface ProductPreviewModalProps {
+	productTitle: string;
+	productVendor: JSX.Element | string | null;
+	productIcon: string;
+	productId: number;
+	triggerRef: React.RefObject< HTMLAnchorElement >;
+	onOpen?: () => void;
+	onClose?: ( closeType?: string ) => void;
+}
+
+export default function ProductPreviewModal( {
+	productTitle,
+	productVendor,
+	productIcon,
+	productId,
+	triggerRef,
+	onOpen,
+	onClose,
+}: ProductPreviewModalProps ) {
+	const [ isLoading, setIsLoading ] = useState( true );
+	const [ previewContent, setPreviewContent ] = useState< {
+		html: string;
+		css: string;
+	} | null >( null );
+	const [ error, setError ] = useState< string | null >( null );
+
+	const closeModal = useCallback(
+		( closeType = '' ) => {
+			if ( onClose ) {
+				onClose( closeType );
+			}
+			// Return focus to the triggering element after a small delay
+			// to ensure layout shifts have completed
+			setTimeout( () => {
+				if ( triggerRef.current ) {
+					triggerRef.current.focus();
+				}
+			}, 100 );
+		},
+		[ onClose, triggerRef ]
+	);
+
+	// Add event listener for content interactions
+	useEffect( () => {
+		const handleContentInteraction = ( event: Event ) => {
+			const target = event.target as HTMLElement;
+			const link = target.closest( 'a' );
+			if ( link ) {
+				const trackType = link.getAttribute( 'data-iam-tracks' );
+				if ( trackType === 'buy_now' ) {
```

Comment:
WDYT of such approach?
```suggestion
				if ( trackType ) {
					closeModal(
						`marketplace_product_preview_modal_${ trackType }_clicked`
					);
				}
```

Reply:
This should cover us if we add other links in the future

Reply:
Wish I'd thought of this...

Done in bf66c9e3ff